kolibri/core/assets/src/core-app/constructor.js

```diff
@@ -25,125 +32,116 @@ const publicMethods = [
-  // libraries
-  this.logging = require('../logging');
-  this.vue = vue;
-  this.vuex = vuex;
-  this.conditionalPromise = require('../conditionalPromise');
-}
-
-/**
  * Constructor for object that forms the public API for the Kolibri
  * core app.
  * @constructor
  */
-module.exports = function CoreApp() {
-  this.lib = new Lib();
-  this.resources = new ResourceManager(this);
-  const mediator = new Mediator();
-
-  this.constants = require('../constants');
-  this.coreActions = require('../core-actions');
-
-  Object.keys(Resources).forEach((resourceClassName) =>
-    this.resources.registerResource(resourceClassName, Resources[resourceClassName]));
-
-  vue.prototype.Kolibri = this;
-  /**
-   * Use vuex for state management.
-   */
-  vue.use(vuex);
-
-  // Register global components
-  vue.component('content-render', require('../vue/content-renderer'));
-  vue.component('download-button', require('../vue/content-renderer/download-button'));
-  vue.component('loading-spinner', require('../vue/loading-spinner'));
-  vue.component('core-modal', require('../vue/core-modal'));
-  vue.component('progress-bar', require('../vue/progress-bar'));
-  vue.component('content-icon', require('../vue/content-icon'));
-  vue.component('core-base', require('../vue/core-base'));
-
-  this.i18n = {
-    reversed: false,
-  };
-
-  // Shim window.location.origin for IE.
-  if (!window.location.origin) {
-    window.location.origin = `${window.location.protocol}//${window.location.hostname}${(
-          window.location.port ? `:${window.location.port}` : '')}`;
-  }
 
-  const self = this;
+module.exports = class CoreApp {
+  constructor() {
+    Object.assign(this, constructorExport());
+
+    this.resources = new ResourceManager(this);
+    const mediator = new Mediator();
 
-  function setUpVueIntl() {
+    Object.keys(Resources).forEach((resourceClassName) =>
+      this.resources.registerResource(resourceClassName, Resources[resourceClassName]));
+
+    vue.prototype.Kolibri = this;
     /**
-     * Use the vue-intl plugin.
-     **/
-    const VueIntl = require('vue-intl');
-    vue.use(VueIntl);
-    vue.prototype.$tr = function $tr(messageId, ...args) {
-      const defaultMessageText = this.$options.$trs[messageId];
-      const message = {
-        id: `${this.$options.$trNameSpace}.${messageId}`,
-        defaultMessage: defaultMessageText,
+     * Use vuex for state management.
+     */
+    vue.use(vuex);
+
+    // Register global components
+    vue.component('content-render', require('../vue/content-renderer'));
+    vue.component('download-button', require('../vue/content-renderer/download-button'));
+    vue.component('loading-spinner', require('../vue/loading-spinner'));
+    vue.component('core-modal', require('../vue/core-modal'));
+    vue.component('progress-bar', require('../vue/progress-bar'));
+    vue.component('content-icon', require('../vue/content-icon'));
+    vue.component('core-base', require('../vue/core-base'));
```

Comment:
I think this makes that process a lot more amenable, but would rather do that refactor in another PR, if it seems right.